crypto/bio/bio.c

```diff
@@ -212,6 +212,10 @@ int BIO_gets(BIO *bio, char *buf, int len) {
     OPENSSL_PUT_ERROR(BIO, BIO_R_UNSUPPORTED_METHOD);
     return -2;
   }
+  if (len <= 0) {
+    return 0;
+  }
```

Comment:
We might also want to consider applying this change to some of the other BIO functions where `len` is used in the callback (i.e. BIO_read and BIO_write)

Reply:
Thanks for the heads up, made the changes for those as well :)